calculation/gmhazard_calc/gmhazard_calc/scenario/scenario.py

```diff
@@ -36,78 +36,37 @@ def run_ensemble_scenario(
     EnsembleScenarioResult
     """
     ims = shared.get_SA_ims(ensemble.ims, component=im_component)
-
     scenario_branches = run_branches_scenario(ensemble, ims, site_info)
 
-    # Combine the branches according to their weights
-    mu, sigma, weights = None, None, []
-    for cur_scenario in scenario_branches:
-        cur_mu = cur_scenario.branch.weight * cur_scenario.mu_data
-        cur_sigma = cur_scenario.branch.weight * cur_scenario.sigma_data
-        weights.append(cur_scenario.branch.weight)
-        if mu is None:
-            mu, sigma = cur_mu, cur_sigma
-
-            # Creates a multi-index dataframe with IM / Rupture as indexes and one column for the branch
-            # Values for this dataframe is the Scenario Branch IM data
-            im_branch_ruptures = pd.concat(
-                [
-                    pd.DataFrame(
-                        val,
-                        index=cur_scenario.mu_data.index.values,
-                        columns=[cur_scenario.branch.name],
-                    )
-                    for val in cur_scenario.mu_data.T.values
-                ],
-                keys=cur_scenario.mu_data,
-            )
-        else:
-            mu = mu.add(cur_mu)
-            sigma = sigma.add(cur_sigma)
-
-            # Creates a multi-index dataframe with IM / Rupture as indexes and one column for the branch
-            # Values for this dataframe is the Scenario Branch IM data
-            # Add another branch to the dataframe as a new column via join
-            im_branch_ruptures = im_branch_ruptures.join(
-                pd.concat(
-                    [
-                        pd.DataFrame(
-                            val,
-                            index=cur_scenario.mu_data.index.values,
-                            columns=[cur_scenario.branch.name],
-                        )
-                        for val in cur_scenario.mu_data.T.values
-                    ],
-                    keys=cur_scenario.mu_data,
-                )
-            )
-
-    # Calculating Percentiles
-    im_branch_ruptures, weights = np.asarray(im_branch_ruptures), np.asarray(weights)
-    weights = np.repeat(weights[None, ...], len(im_branch_ruptures), 0)
-
-    # Sorting
-    sort_ind = np.argsort(im_branch_ruptures, axis=1)
-    im_branch_ruptures = np.take_along_axis(im_branch_ruptures, sort_ind, 1)
-    weights = np.take_along_axis(weights, sort_ind, 1)
-
-    # Inverse CDF lookup
-    cdf_x, cdf_y = im_branch_ruptures, np.cumsum(weights, axis=1)
-    x_values = sha_calc.shared.query_non_parametric_multi_cdf_invs(
-        [0.16, 0.5, 0.84], cdf_x, cdf_y
+    # Calculate mu
+    mu = sum(
+        cur_scenario.branch.weight * cur_scenario.mu_data
+        for cur_scenario in scenario_branches
```

Comment:
I'm pretty sure this should be log mean. Probably worth checking.

Reply:
Yeah, I was confused too...

Does this `mu` indicate the weighted average?

Reply:
This makes sense. It means that the IM values would be consistent with what we do in the hazard curves.
Joel and I looked at whether the hazard branch combining used ln mean or not and it appeared to not but if the IM_values were already in log space that would make sense.